python/cuda_parallel/tests/test_segmented_reduce.py

```diff
@@ -4,31 +4,41 @@
 
 import cupy as cp
 import numpy as np
+import pytest
 
 import cuda.parallel.experimental.algorithms as algorithms
 from cuda.parallel.experimental.struct import gpu_struct
 
 
-def test_segmented_reduce(input_array):
-    "Test for all supported input types"
+@pytest.fixture(params=["i4", "u4", "i8", "u8"])
+def offset_dtype(request):
+    return np.dtype(request.param)
+
+
+def test_segmented_reduce(input_array, offset_dtype):
+    "Test for all supported input types and for some offset types"
 
     def binary_op(a, b):
         return a + b
 
     assert input_array.ndim == 1
     sz = input_array.size
     rng = np.random.default_rng()
-    n_segments = 2**4
+    n_segments = 16
     h_offsets = np.zeros(n_segments + 1, dtype="int64")
-    h_offsets[1:] = rng.multinomial(sz, [1 / 16] * 16)
+    h_offsets[1:] = rng.multinomial(sz, [1 / n_segments] * n_segments)
```

Comment:
We could do this on GPU

Reply:
Done! In this example the value of `n_segments` is small, so populating it on GPU might actually be slower than doing so on a GPU and copying, but I think the change to do it on the GPU is the right thing to do. First, to set example, second to permit user to scale up `n_segments` and `sz` values without needing to change the code much.